### ISSUE TITLE
Potentially half-bricking the system when touching VLan/Bridge (With Fix!)

### ISSUE THREAD

**Issue by windhamwong** [opnsense/core#2052]
Hello, I recently updating the vlan and bridge config via Web UI. The system is half-bricked after a reboot. It allows me to do console login via monitor/keyboard only. I still cannot figure out what is the cause of bricking at the first place, but I found out the cause of being broken after reboot.

**Details:**
Version: **17.7**
Updating VLan and Bridge config via WebUI and rebooting causing half-bricking the system.

**Symptom:**
No more internet access from LANs (not WAN) and no SSH/Web UI access.
Console available only (Monitor/Keyboard)
Running command `/usr/local/etc/rc.restart-webgui` gives PHP error of:

```
Warning: require_once(config.inc): failed to open stream: No such file or directory in /usr/local/etc/rc.restart_webgui on line 4

Fatal error: require_once(): Failed opening required 'config.inc' (include_path='.:/usr/local/share/pear') in /usr/local/etc/rc.restart_webgui on line 4
```

**Findings:**
- The error appears to be PHP error. Suspecting the `include_path` isn't loaded correctly.
- pf*ense has the same issue since 2007 but no fix or update in the post. ref: https://forum.pf*ense.org/index.php?topic=6572.0

As the script is written in PHP, that means it should load include_path from `php.ini`. The default php file loaded by the system is `/usr/local/etc/php.ini`. This file should be generated by `configd` but somehow the file appears empty and nothing has been loaded or generated.
After a deeper digging into the system, I found out that adding this line (generated by `configd`),
`include_path = ".:/usr/local/etc/inc:/usr/local/www:/usr/local/opnsense/mvc:/usr/local/share/pear:/usr/local/opnsense/contrib"`
can fix the problem. A full reboot is required after this modification.

However, as I still don't know how this is triggered at the first place, might need developers to look into this one.

**Steps for fixing:**
1. Console log onto the system
2. Try to run `/usr/local/etc/rc.restart-webgui` to see if you get the same error as I mentioned. (If Yes, continue)
3. Adding the line, 
`include_path = ".:/usr/local/etc/inc:/usr/local/www:/usr/local/opnsense/mvc:/usr/local/share/pear:/usr/local/opnsense/contrib"` into `/usr/local/etc/php.ini`.
4. Try to run again `/usr/local/etc/rc.restart-webgui`. The script should run.
5. Run `/usr/local/etc/rc.initial` and select Reboot.
6. Your box is back!

**Comment by fichtner** [comment]
Your filesystem is unstable and will either not write php.ini or bring back an empty copy after reboot. This is not really related to reconfiguring, more likely caused by a kernel crash and forced reboot during the configuring.

The thing is the system will even write php.ini on boot, but that seems defective as well, which is strange, because that is issued by a shell script prior to PHP-based startup sequence.

18.1 will have a firmware health audit tool to see if your install is healthy or has damaged files that could prevent the system from functioning normally.

I'm entirely unsure what to do. php.ini can and will disappear on an UFS-based filesystem and we have a recovery option for that which seems to fail as well for you.

**Comment by fichtner** [comment]
PS, the health check in a nutshell is running these two commands on the root console:

    # pkg check -da
    # pkg check -sa

**Comment by windhamwong** [comment]
Good to hear about news in 18.1. This is weird and rare as well, and no solution found online. I can see the php.ini is generated during the startup sequence but somehow I can see the startup script stops running after setting up root (not my custom user account) and files (not php.ini). During my investigation, `/etc/passwd` is not updated. This might help for further investigation in the startup sequence.

**Comment by fichtner** [comment]
ok, let's see. here is where the php.ini is generated during bootup in the early /usr/local/etc/rc sequence:

https://github.com/opnsense/core/blob/master/src/etc/rc#L172

Amongst others it starts these two scripts in this order:

https://github.com/opnsense/core/blob/master/src/etc/rc.syshook.d/10-configd.early
https://github.com/opnsense/core/blob/master/src/etc/rc.syshook.d/15-php.early

So from these two you can try to figure out which one fails and why it leads to improper php.ini setup :)

**Comment by fichtner** [comment]
A full dump of the failing boot sequence output may help shed light on this as well.

**Comment by windhamwong** [comment]
@fichtner do you have the path of the log?

**Comment by fichtner** [comment]
no, just whatever the primary console outputs during bootup

**Comment by windhamwong** [comment]
oh.. I don't have it because I asked someone to do it for me remotely...
would syslog helps?

**Comment by fichtner** [comment]
no, since this fails prior to syslog start

**Comment by windhamwong** [comment]
I couldn't help then because I have already fixed it :(

**Comment by fichtner** [comment]
No worries, you detailed description may help others in the future. Shall we close this then until we have new data?

**Comment by windhamwong** [comment]
Closing this. Reopening when more data comes in.
